### PR TITLE
[FLINK-22063][table] Fix bug that some join conditions are lost when …

### PR DESCRIPTION
**Pull request by cshuo**
…building Lookup Join ExecNode

## What is the purpose of the change

Fix bug that some join conditions are lost when building remaining condition of LookupJoin.

Root  Cause:
Considering the join key pairs are `l.a = r.e and l.b = r.e`, we use `Map<lookup_key, left_key>` as `allLookupKeys` information, because lookup keys should be deduplicated. In this case we get `<r.e, l.a>`. 

When building remaining conditions, the expected remaining condition should be `l.b = r.e`, but currently, we check whether the right key of key-pairs  is in `allLookupKeys`, which leading to an empty result.  This can be fixed by checking left key of 
key-pairs instead.

## Brief change log
- Fix bug of `CommonPhysicalLookupJoin` when building `remainingCondition`

## Verifying this change
- Add plan test.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by cshuo** [comment]
@leonardBang Please have a look at this.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/common/CommonPhysicalLookupJoin.scala [hunk 1]
```diff
@@ -109,7 +109,7 @@ abstract class CommonPhysicalLookupJoin(
     cluster.getRexBuilder,
     input.getRowType,
     calcOnTemporalTable,
-    allLookupKeys.keys.toList.sorted.toArray,
+    allLookupKeys.values.toList,
```
Comment (leonardBang) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/common/CommonPhysicalLookupJoin.scala:112]: minor: could you help update the comment to 
` remaining condition used to filter the joined records (left input record X lookup-ed records)`

**Review by leonardBang — APPROVED** [review]
@cshuo Thanks for the fix, the PR looks good, could you help update the note and rebase to latest mater?

**Comment by leonardBang** [comment]
The failed test is not related to this PR, cc @JingsongLi could you help merge?

**Comment by JingsongLi** [comment]
@cshuo Please add some comments to explain why

**Review by JingsongLi — APPROVED** [review]
Looks good to me.

**Pull request merged**